target/ppc/dfp_helper.c

```diff
@@ -1391,3 +1391,68 @@ DFP_HELPER_SHIFT(DSCLI, 64, 1)
 DFP_HELPER_SHIFT(DSCLIQ, 128, 1)
 DFP_HELPER_SHIFT(DSCRI, 64, 0)
 DFP_HELPER_SHIFT(DSCRIQ, 128, 0)
+
+target_ulong helper_CDTBCD(target_ulong s)
+{
+    uint64_t res = 0;
+    uint32_t dec32, declets;
+    uint8_t bcd[6];
+    int i, w, sh;
+    decNumber a;
+
+    for (w = 1; w >= 0; w--) {
+        res <<= 32;
+        declets = extract64(s, 32 * w, 20);
+        if (declets) {
+            /* decimal32 with zero exponent and word "w" declets */
+            dec32 = (0x225ULL << 20) | declets;
+            decimal32ToNumber((decimal32 *)&dec32, &a);
```

Comment:
Creating a decNumber seems unnecessary, it's better to use `DPD2BCD` from `libdecnumber/decDPD.h`

Reply:
As the commit message says, DPD2BCD is not available. We'd need to define `DEC_DPD2BCD` before including `libdecnumber/decDPD.h`, and that would add a 2kb lookup table just for this instruction.

Reply:
Good point, looking at the code in `include/libdecnumber/` and `libdecnumber/` I couldn't find some way to access the table without some code in weird places where it didn't fit well or some strange workaround